packages/common/src/services/audius-backend/AudiusBackend.ts

```diff
@@ -387,35 +387,50 @@ export const audiusBackend = ({
         }
       )
     }
+    const start = preloadImageTimer.start()
+    const timeoutMs =
+      getRemoteVar(IntKeys.IMAGE_QUICK_FETCH_TIMEOUT_MS) ?? undefined
+    let timeoutId: Nullable<NodeJS.Timeout> = null
 
-    return new Promise<string | false>((resolve) => {
-      const start = preloadImageTimer.start()
+    try {
+      const response = await Promise.race([
+        fetch(url),
+        new Promise<Response>((_, reject) => {
+          timeoutId = setTimeout(() => reject(new Error('Timeout')), timeoutMs)
+        })
+      ])
+
+      if (timeoutId) {
+        clearTimeout(timeoutId)
+      }
 
-      const timeoutMs =
-        getRemoteVar(IntKeys.IMAGE_QUICK_FETCH_TIMEOUT_MS) ?? undefined
-      const timeout = setTimeout(() => {
-        preloadImageTimer.end(start)
-        resolve(false)
-      }, timeoutMs)
+      if (!response.ok) {
+        return false
+      }
 
-      // Avoid garbage collection by keeping a few images in an in-mem array
+      const blob = await response.blob()
+      const objectUrl = URL.createObjectURL(blob)
```

Comment:
should we remove this and just do `image.src = url` below?

Reply:
no cause then it would fetch again? or are you thinking since fetch already happened it wouldn't?

Reply:
ah nvm I think I was misunderstanding what this does here. this looks good!